src/DGtal/dec/ATSolver2D.h

```diff
@@ -0,0 +1,1028 @@
+                            Minimum, ///< compute minimum value at cell vertices,
+                            Maximum, ///< compute maximum value at cell vertices
+    };
+    typedef typename KSpace::Space                               Space;
+    typedef typename Space::RealVector                           RealVector;
+    typedef typename RealVector::Component                       Scalar;
+    typedef typename KSpace::SCell                               SCell;
+    typedef typename KSpace::Cell                                Cell;
+    typedef typename KSpace::Surfel                              Surfel;
+    typedef HyperRectDomain<Space>                               Domain;
+    typedef DiscreteExteriorCalculus<2,dimension, LinearAlgebra> Calculus;
+    typedef typename KSpace::template SurfelMap<double>::Type    SmallestEpsilonMap;
+    typedef typename Calculus::Index                             Index;
+    typedef typename Calculus::PrimalForm0                       PrimalForm0;
+    typedef typename Calculus::PrimalForm1                       PrimalForm1;
+    typedef typename Calculus::PrimalForm2                       PrimalForm2;
+    typedef typename Calculus::PrimalIdentity0                   PrimalIdentity0;
+    typedef typename Calculus::PrimalIdentity1                   PrimalIdentity1;
+    typedef typename Calculus::PrimalIdentity2                   PrimalIdentity2;
+    typedef typename Calculus::PrimalDerivative0                 PrimalDerivative0;
+    typedef typename Calculus::PrimalDerivative1                 PrimalDerivative1;
+    typedef typename Calculus::PrimalAntiderivative1             PrimalAntiderivative1;
+    typedef typename Calculus::PrimalAntiderivative2             PrimalAntiderivative2;
+    typedef typename Calculus::PrimalHodge0                      PrimalHodge0;
+    typedef typename Calculus::PrimalHodge1                      PrimalHodge1;
+    typedef typename Calculus::PrimalHodge2                      PrimalHodge2;
+    typedef typename KSpace::template SurfelMap<Index>::Type     Surfel2IndexMap;
+
+    // SparseLU is so much faster than SparseQR
+    // SimplicialLLT is much faster than SparseLU
+    // SimplicialLDLT is as fast as SimplicialLLT but more robust
+    // typedef EigenLinearAlgebraBackend::SolverSparseQR LinearAlgebraSolver;
+    // typedef EigenLinearAlgebraBackend::SolverSparseLU LinearAlgebraSolver;
+    // typedef EigenLinearAlgebraBackend::SolverSimplicialLLT LinearAlgebraSolver;
+    typedef EigenLinearAlgebraBackend::SolverSimplicialLDLT LinearAlgebraSolver;
+    typedef DiscreteExteriorCalculusSolver<Calculus, LinearAlgebraSolver, 2, PRIMAL, 2, PRIMAL> SolverU2;
+    typedef DiscreteExteriorCalculusSolver<Calculus, LinearAlgebraSolver, 0, PRIMAL, 0, PRIMAL> SolverV0;
+
+    /// A smart (or not) pointer to a calculus object.
+    CountedConstPtrOrConstPtr< Calculus > ptrCalculus;
+    /// the derivative operator for primal 0-forms
+    PrimalDerivative0     primal_D0;
```

Comment:
shouldn't all these be `private:` members ?

Reply:
ok